### DISCUSSION TITLE
Mesh resolution

### DISCUSSION THREAD

**Discussion by kasia129** [capytaine/capytaine#117] [Q&A] [answered]
Can I ask about the reason behind this condition in the solve method of BEMSolver:

```
        if problem.wavelength < 8*problem.body.mesh.faces_radiuses.max():
            LOG.warning(f"Resolution of the mesh (8×max_radius={8*problem.body.mesh.faces_radiuses.max():.2e}) "
                        f"might be insufficient for this wavelength (wavelength={problem.wavelength:.2e})!")
```

I understand face_radiuses.max is the max of the distances between the centres and vertices of each panel, so that represents some characteristic size of the panels. But where does the factor 8 come from, and why, from solver/hydrodynamics point of view, is this condition necessary?

Since I'm using AxialSymmetricMesh.from_profile for meshing, the panels on the top and bottom of my body remain large, no matter what the radial resolution nphi, as per the sample plot below:

![image](https://user-images.githubusercontent.com/76471057/143480724-51dff1b4-4be8-44a8-bc6c-504be63dca7f.png)

**Comment by mancellin** [comment, marked as answer]
Having at least a few panels per wavelength is more or less Nyquist-Shannon sampling theorem.
If I remember correctly, the number 8 comes from a recommendation in the documentation of Nemoh.

You might be able to have a finer mesh by adding more points to the profile provided to `AxialSymmetricMesh`.

**Reply by kasia129** [reply]
Thanks, I managed to refine the mesh this way and now it seems to work fine.
